vision_stack/task_buoy/src/task_buoy_new.cpp

```diff
@@ -0,0 +1,89 @@
+#include <ros/ros.h>
+#include <actionmsg/buoyAction.h>
+#include <kraken_msgs/scanAction.h>
+#include <actionlib/client/simple_action_client.h>
+#include <actionlib/client/terminal_state.h>
+#include <task_buoy/resultheader.h>
+#include <task_buoy/taskheader.h>
+
+#include <resources/topicHeader.h>
+#include <kraken_msgs/absoluteRPY.h>
+#include <kraken_msgs/setYawAction.h>
+
+#include <signal.h>
+
+typedef actionlib::SimpleActionClient<actionmsg::buoyAction> Client;
+typedef actionlib::SimpleActionClient<kraken_msgs::setYawAction> Client1;
+typedef actionlib::SimpleActionClient<kraken_msgs::scanAction> Client2;
+
+double current_yaw;
+
+void getyawCallback(const kraken_msgs::absoluteRPY& msg)
+{
+    current_yaw = msg.yaw;
+}
+
+void my_handler(sig_t s){
+           printf("Caught signal %d\n",s);
+           exit(1);
+
+}
+
+int main(int argc, char ** argv)
+{
+    ros::init(argc, argv, "buoy_node");
+    //signal (SIGINT,my_handler);
```

Comment:
Was getting a compilation error here. Commented this out temporarily. Have to fix this so that we can preempt the rotation and resume it by just stopping and starting the client.